src/loaders/video_loader.py

```diff
@@ -33,10 +35,29 @@ def load(self):
                 _, frame = video.read()
                 continue
 
+            # Skip similar frames if threshold is a positive value
```

Comment:
Again don't change the SimpleVideoLoader, it is meant to remain the way it is. right now I'm now even sure if we want your logic to be tightly coupled with a video loader. I mean we may even want to use frame differencing logic somewhere else during the pipeline. Shouldn't it be a PP(probabilistic predicate)? Just an opinion, do let me know if you guys have some other views.